server/controllers/UsersController.js

```diff
@@ -98,6 +111,92 @@ class UsersController {
         }
       }));
   }
+
+  /**
+   * This controller updates a users profile
+   * @param {object} req
+   * @param {object} res
+   * @param {object} next
+   * @returns {void}
+   */
+  static updateProfile(req, res, next) {
+    // get token from request header
+    const token = req.headers.authorization;
+    if (!token) {
```

Comment:
Great job. But I feel like this part checking for and verifying a token should be a middleware. So that everyone else can use it for creating protected routes.

Reply:
Oh yes. That will be useful, I'll implement that.